Userland/Applications/Browser/BookmarksBarWidget.cpp

```diff
@@ -1,117 +1,33 @@
 #include <LibGUI/Event.h>
 #include <LibGUI/JsonArrayModel.h>
 #include <LibGUI/Menu.h>
 #include <LibGUI/MessageBox.h>
 #include <LibGUI/Model.h>
-#include <LibGUI/TextBox.h>
 #include <LibGUI/Widget.h>
-#include <LibGUI/Window.h>
 #include <LibGfx/Palette.h>
 
 namespace Browser {
 
-namespace {
-
-enum class PerformEditOn {
-    NewBookmark,
-    ExistingBookmark
-};
-
-class BookmarkEditor final : public GUI::Dialog {
-    C_OBJECT(BookmarkEditor)
-
-public:
-    static Vector<JsonValue>
-    edit_bookmark(Window* parent_window, StringView title, StringView url, PerformEditOn perform_edit_on)
-    {
-        auto editor = BookmarkEditor::construct(parent_window, title, url);
-        if (perform_edit_on == PerformEditOn::NewBookmark) {
-            editor->set_title("Add Bookmark");
-        } else {
-            editor->set_title("Edit Bookmark");
-        }
-        editor->set_icon(g_icon_bag.bookmark_filled);
-
-        if (editor->exec() == ExecResult::OK) {
-            return Vector<JsonValue> { editor->title(), editor->url() };
-        }
-
-        return {};
-    }
-
-private:
-    BookmarkEditor(Window* parent_window, StringView title, StringView url)
-        : Dialog(parent_window)
-    {
-        auto widget = set_main_widget<GUI::Widget>();
-        widget->load_from_gml(edit_bookmark_gml).release_value_but_fixme_should_propagate_errors();
-
-        set_resizable(false);
-        resize(260, 85);
-
-        m_title_textbox = *widget->find_descendant_of_type_named<GUI::TextBox>("title_textbox");
-        m_title_textbox->set_text(title);
-        m_title_textbox->set_focus(true);
-        m_title_textbox->select_all();
-
-        auto& ok_button = *widget->find_descendant_of_type_named<GUI::Button>("ok_button");
-        ok_button.on_click = [this](auto) {
-            done(ExecResult::OK);
-        };
-        ok_button.set_default(true);
-
-        m_url_textbox = *widget->find_descendant_of_type_named<GUI::TextBox>("url_textbox");
-        m_url_textbox->set_text(url);
-        m_url_textbox->on_change = [this, &ok_button]() {
-            auto has_url = !m_url_textbox->text().is_empty();
-            ok_button.set_enabled(has_url);
-        };
-
-        auto& cancel_button = *widget->find_descendant_of_type_named<GUI::Button>("cancel_button");
-        cancel_button.on_click = [this](auto) {
-            done(ExecResult::Cancel);
-        };
-    }
-
-    ByteString title() const
-    {
-        return m_title_textbox->text();
-    }
-
-    ByteString url() const
-    {
-        return m_url_textbox->text();
-    }
-
-    RefPtr<GUI::TextBox> m_title_textbox;
-    RefPtr<GUI::TextBox> m_url_textbox;
-};
-
-}
-
 static BookmarksBarWidget* s_the;
 
 BookmarksBarWidget& BookmarksBarWidget::the()
 {
     return *s_the;
 }
 
-BookmarksBarWidget::BookmarksBarWidget(ByteString const& bookmarks_file, bool enabled)
+BookmarksBarWidget::BookmarksBarWidget(ByteString const& bookmarks_file, bool const enabled)
```

Comment:
This (and most of the changes in this commit) don't seem to be related to GML compilation. Please split them out into a separate commit or drop them completely.